terraform/gcp-kubernetes-cluster/vault.tf

```diff
@@ -0,0 +1,114 @@
+# Create a service account for the Vault KMS
+resource "google_service_account" "vault_kms_service_account" {
+  account_id   = "${var.cluster_name}-vault-gcpkms"
+  display_name = "Vault KMS for auto-unseal"
+}
+
+# This will be used to create a credentials.json inside a k8s secret obj
+resource "google_service_account_key" "vault_kms_service_account_key" {
+  service_account_id = google_service_account.vault_kms_service_account.name
+}
+
+# Create a KMS Key
+resource "google_kms_key_ring" "key_ring" {
+  name     = "vault-keyring-gcp"
+  location = var.region
+}
+
+# Create a crypto key for the key ring. This key will be used to automatically unseal Vault
+resource "google_kms_crypto_key" "crypto_key" {
+  name            = "vault-cryptokey-gcp"
+  key_ring        = google_kms_key_ring.key_ring.id
+  rotation_period = "100000s"
+}
+
+# We give vault service account access to that key ring. Making im owner of the key.
+resource "google_kms_key_ring_iam_binding" "vault_iam_kms_binding" {
+  key_ring_id = google_kms_key_ring.key_ring.id
+  role        = "roles/owner"
+
+  members = [
+    "serviceAccount:${google_service_account.vault_kms_service_account.email}",
+  ]
+}
+
+# Vault certificates
+resource "tls_private_key" "vault_key" {
+  algorithm = "RSA"
+  rsa_bits  = 2048
+}
+
+# This requires that jq and openssl are installed in the runtime environment
+# It creates a certificate signing request (CSR) based on the vault-csr.conf file
+# The 2 jq at the beginning and end of the pipes are used to read the input and wrap the result in json
+# since this is how terraform "external" passes data.
+data "external" "k8s_cert_request" {
+  program = [
+    "bash", "-c",
+    "curl -o jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && chmod +x jq && jq -rc '.key' | openssl req -new -noenc -config ${path.module}/vault-csr.conf -key /dev/stdin | jq -rRncs '{\"request\": inputs}'"
```

Comment:
isn't sideloading dependencies like this from arbitrary URL dangerous? why can't we install this as a package in the runtime environment instead?

Reply:
We could  do that

Reply:
Finally, I just checked, and we need this `data` resource to create [the certificate](https://github.com/ai-cfia/infra/blob/1-create-a-kubernetes-cluster-for-gcp-w-the-ingress-nginx/terraform/gcp-kubernetes-cluster/vault.tf#L62). I looked on the Terraform marketplace to see if there is a viable `jq` provider, but the majority doesn't seem to be maintained anymore. I have opened [this issue]() to investigate the problem in the futur.